bitbots_localization/src/MotionModel.cpp

```diff
@@ -18,16 +18,28 @@ RobotMotionModel::RobotMotionModel(const particle_filter::CRandomNumberGenerator
 }
 
 void RobotMotionModel::drift(RobotState &state,
-                             geometry_msgs::Vector3 movement,
-                             geometry_msgs::Vector3 movement2) const {
-  // sample_motion_model_odometry
-  double deltaRot1New = movement.x - sample(0.001 * abs(movement.x) + 0.001 * movement.y);
-  double deltaTransNew = movement.y - sample(0.001 * movement.y + 0.001 * (abs(movement.x) + abs(movement.z)));
-  double deltaRot2New = movement.z - sample(0.001 * abs(movement.z) + 0.001 * movement.y);
-
-  state.setXPos(state.getXPos() + deltaTransNew * cos(state.getTheta() + deltaRot1New));
-  state.setYPos(state.getYPos() + deltaTransNew * sin(state.getTheta() + deltaRot1New));
-  double theta = state.getTheta() + deltaRot1New + deltaRot2New;
+                             geometry_msgs::Vector3 linear_movement,
+                             geometry_msgs::Vector3 rotational_movement) const {
+
+  // Convert cartesian coordinates to polarcoordinates with an orientation
+  auto [polar_rot, polar_dist] = cartesianToPolar(linear_movement.x, linear_movement.y);
+  double orientation = rotational_movement.z - polar_rot;
+
+  // Apply sample drift for odom data
+  double deltaRot1_with_offset = polar_rot - sample(0.001 * abs(polar_rot) + 0.001 * polar_dist);
```

Comment:
I would prefer `_with_drift` instead of `_with_offset`.

Reply:
Seems better